odoo/orm/domains.py

```diff
@@ -1111,6 +1111,25 @@ def _optimize_in_required(condition, model):
     return DomainCondition(condition.field_expr, condition.operator, value)
 
 
+@operator_optimization(['any', 'not any'], OptimizationLevel.FULL)
+def _optimize_bypass_comodel_id_lookup(condition, model):
+    """Avoid comodel's subquery, if it can be compared with the field directly"""
+    field = condition._field(model)
+    if (
+        model.env.is_superuser()
+        and '.' not in condition.field_expr
+        and field.relational
+        and not field.auto_join
```

Comment:
Why avoid auto_join fields?
auto_join means bypass permission, so it's equivalent to su in this context.